src/Simulation/QSharpFoundation/Bitwise/Bitwise.cs

```diff
@@ -49,11 +49,13 @@ public class Native : Parity
             static long ParityFunc(long val)
             {
                 // parity function using idea described at http://graphics.stanford.edu/~seander/bithacks.html#ParityMultiply
-                ulong v = System.Convert.ToUInt64(val);
+                ulong v = unchecked((ulong)val);
```

Comment:
Does this line really needs to be unchecked? It seems like without any modification it could be casted to `ulong`, even using `System.Convert.ToUInt64`.

Reply:
Using [System.Convert.ToUInt64()](https://docs.microsoft.com/en-us/dotnet/api/system.convert.touint64?view=net-6.0#system-convert-touint64(system-int64)) it throws (if arg is negative), see the bug in the Description of this PR.
Using explicit cast with `(ulong)` it may or may not throw depending on the [CheckForOverflowUnderflow compiler option](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/checked-and-unchecked).
With `unchecked()` I expect it to never throw.